File: multifiledownloader/src/main/java/com/github/why168/multifiledownloader/DownloadManager.java
package com.github.why168.multifiledownloader;

import android.content.Context;
import android.content.Intent;

/**
 * DownloadManager
 *
 * @author Edwin.Wu
 * @version 2017/1/16 15:27
 * @since JDK1.8
 */
public class DownloadManager {
    private static DownloadManager instance;
    private final Context context;

    private DownloadManager(Context context) {
        this.context = context;
        context.startService(new Intent(context, DownLoadService.class));
    }

    public static DownloadManager getInstance(Context context) {
        if (instance == null) {
            synchronized (DownloadManager.class) {
                if (instance == null) {
                    instance = new DownloadManager(context);
                }
            }
        }
        return instance;
    }


    public void down(DownLoadBean item) {
        Intent intent = new Intent(context, DownLoadService.class);
        intent.setAction("com.github.why168.multifiledownloader.downloadservice");
        intent.setPackage("com.github.why168.multifiledownloader");
        intent.putExtra(Constants.KEY_DOWNLOAD_ENTRY, item);
        intent.putExtra(Constants.KEY_OPERATING_STATE, false);
        context.startService(intent);
    }

    public void delete(DownLoadBean item) {
        Intent intent = new Intent(context, DownLoadService.class);
        intent.setAction("com.github.why168.multifiledownloader.downloadservice");
        intent.setPackage("com.github.why168.multifiledownloader");
        intent.putExtra(Constants.KEY_DOWNLOAD_ENTRY, item);
        intent.putExtra(Constants.KEY_OPERATING_STATE, true);
        context.startService(intent);
    }
}

File: multifiledownloader/src/main/java/com/github/why168/multifiledownloader/DownloadManager.kt
package com.github.why168.multifiledownloader

import android.content.Context
import android.content.Intent
import java.lang.ref.SoftReference

/**
 * DownloadManager
 *
 * @author Edwin.Wu
 * @version 2017/1/16 15:27
 * @since JDK1.8
 */
class DownloadManager {

    private var context: Context? = null

    companion object {

        @Volatile
        private var instance: SoftReference<DownloadManager>? = null

        fun getInstance(context: Context): DownloadManager? {
            if (instance == null && instance?.get() == null) {
                synchronized(DownloadManager::class.java) {
                    if (instance == null && instance?.get() == null) {
                        instance = SoftReference<DownloadManager>(DownloadManager(context))
                    }
                }
            }
            return instance?.get()
        }
    }

    private constructor() {}

    private constructor(context: Context) {
        this.context = context
        context.startService(Intent(context, DownLoadService::class.java))
    }

    fun addTask(item: DownLoadBean) {
        context?.let {
            DownLoadService.addTask(it, item)
        }
    }

    fun deleteTask(item: DownLoadBean) {
        context?.let {
            DownLoadService.deleteTask(it, item)
        }
    }

    fun stopAll() {
        DownLoadService.waitingQueue.clear()

        DownLoadService.connectionTaskMap.forEach {
            it.value.cancel()
        }

        DownLoadService.downTaskMap.forEach {
            it.value.cancel()
        }
    }
}
